junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/DefaultPreInterruptCallback.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2023 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import java.util.Map;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.PreInterruptCallback;
+import org.junit.jupiter.engine.Constants;
+
+/**
+ * The default implementation for {@link PreInterruptCallback},
+ * which will print the stacks of all {@link Thread}s to {@code System.out}.
+ *
+ * <p>Note: This is disabled by default, and must be enabled with
+ * {@link Constants#EXTENSIONS_DEFAULT_PRE_INTERRUPT_CALLBACK_ENABLED_PROPERTY_NAME}
+ *
+ * @since 5.11
+ */
+public class DefaultPreInterruptCallback implements PreInterruptCallback {
```

Comment:
I think `PreInterruptThreadDumpPrinter` or something along those lines would be clearer.